appraisal/ruby-3.4.rb

```diff
@@ -88,7 +88,12 @@
 build_coverage_matrix('elasticsearch', 7..8)
 
 appraise 'relational_db' do
-  gem 'activerecord', '~> 7'
+  # ActiveRecord locked because tests are failing with 7.1, which was attempted as a part of Ruby 3.4 testing in CI.
+  # All existing relational_db configurations use ActiveRecord 7.0. ActiveRecord 7.1 is tested as a part of rails71
+  # appraisal configurations.
+  #
+  # TODO: Repair the failures with ActiveRecord 7.1 in this configuration and remove the locked version.
+  gem 'activerecord', '~> 7.0.0'
```

Comment:
This is not actually a downgrade because all existing relational_db configurations were resolved to AR 7.0.

It also appears that the only Rails version that is tested with database adapters is 6.1. None of 7.x versions are tested. Looks like testing 7.1 with database adapters would fail CI as we currently have it.

Reply:
That is fine!